src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const currentAction of actions) {
+    if (currentAction.type === 'addProperties') {
+      Object.assign(state, currentAction['extraData']);
+    }
+
+    if (currentAction.type === 'removeProperties') {
+      const listToDelete = currentAction['keysToRemove'];
+
+      for (let i = 0; i < listToDelete.length; i++) {
+        const currentDeleteKey = listToDelete[i];
+
+        if (state.hasOwnProperty(currentDeleteKey)) {
+          delete state[currentDeleteKey];
+        }
+      }
+    }
+
+    if (currentAction.type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
+    }
+  }
+
+  return state;
```

Comment:
In task didn`t say about return something, i think without this return code also will work